src/lando/api/legacy/transplants.py

```diff
@@ -356,26 +568,71 @@ def warning_code_freeze(*, repo, **kwargs):
 
 
 @RevisionWarningCheck(9, "Revision has unresolved comments.")
-def warning_unresolved_comments(*, phab, revision, **kwargs):
+def warning_unresolved_comments(
+    revision: dict, diff: dict, stack_state: StackAssessmentState
+):
     if not all(
-        phab.expect(inline, "fields", "isDone")
-        for inline in get_inline_comments(phab, f"D{revision['id']}")
+        stack_state.phab.expect(inline, "fields", "isDone")
+        for inline in get_inline_comments(stack_state.phab, f"D{revision['id']}")
     ):
         return "Revision has unresolved comments."
 
 
-def user_block_no_auth0_email(*, lando_user, **kwargs):
+@RevisionWarningCheck(10, "Revision has multiple authors.")
+def warning_multiple_authors(
+    revision: dict, diff: dict, stack_state: StackAssessmentState
+):
+    """Warn the landing user when a revision has updates from multiple authors."""
+    revision_phid = PhabricatorClient.expect(revision, "phid")
+
+    # Get the author PHID for each diff associated with this revision.
+    author_phids = {
+        PhabricatorClient.expect(diff, "fields", "authorPHID")
+        for diff in stack_state.stack_data.diffs.values()
+        if PhabricatorClient.expect(diff, "fields", "revisionPHID") == revision_phid
+    }
+
+    if len(author_phids) > 1:
+        author_usernames = sorted(
+            reviewer_identity(
+                author_phid, stack_state.users, stack_state.projects
+            ).identifier
+            for author_phid in author_phids
+        )
+        return f"Revision has multiple authors: {', '.join(author_usernames)}."
+
+
+def blocker_user_no_auth0_email(
+    stack_state: StackAssessmentState,
+) -> Optional[str]:
     """Check the user has a proper auth0 email."""
+    if not stack_state.landing_assessment:
+        return None
+
     return (
         None
-        if lando_user.email
+        if stack_state.landing_assessment.auth0_user.email
```

Comment:
@cgsheeh FYI after this merge, `lando_user` got renamed back to `auth0_user` and was not updated in the fixup PR. Not a blocker, but it should be renamed back to `lando_user` everywhere for consistency as this convention is used elsewhere, and references to `auth0_user` were entirely removed from the repo.